### PR TITLE
Moved SonarAnalyzer imports and references out of version props file

### PR DESCRIPTION
**Pull request by duncanp-sonar**
Prep for #2246

### CODE REVIEW

File: src/EmbeddedSonarAnalyzer.props [hunk 1]
```diff
@@ -7,17 +7,4 @@
     <EmbeddedSonarCFamilyAnalyzerVersion>6.17.0.27551</EmbeddedSonarCFamilyAnalyzerVersion>
   </PropertyGroup>
 
-  <ItemGroup Label="Embedded SonarLint analyzers">
```
Comment (duncanp-sonar) [inline src/EmbeddedSonarAnalyzer.props:10]: Previously this props file was only used by Integration.Vsix and it's tests.

Going forward, a property describing the version of the SonarJS analyzer to embed will be added, so the props file will be referenced by the new additional files VSIX.

This change moves the Integration.Vsix-specific bits out of this file.

File: src/Integration.Vsix.UnitTests/Integration.Vsix.UnitTests.csproj [hunk 2]
```diff
@@ -5,7 +5,6 @@
     <RequiresVCProjectEngine>true</RequiresVCProjectEngine>
   </PropertyGroup>
   <Import Project="..\SonarLint.VSSpecificAssemblies.props" />
-  <Import Project="..\EmbeddedSonarAnalyzer.props" />
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/Integration.Vsix.UnitTests.csproj:8]: Moved down a few lines, next to the newly-added ItemGroup.

File: src/Integration.Vsix.UnitTests/Integration.Vsix.UnitTests.csproj [hunk 3]
```diff
@@ -70,6 +69,14 @@
     
   </ItemGroup>
 
+  <!-- Only need to reference the analyzer common assembly for testing. -->
+  <Import Project="..\EmbeddedSonarAnalyzer.props" />
+  <ItemGroup Label="Embedded SonarLint analyzers">
+    <!-- Reference the packages we want to import, but we don't want to include them as analyzers for project -->
+    <PackageReference Include="SonarAnalyzer.CSharp" Version="$(EmbeddedSonarAnalyzerVersion)" ExcludeAssets="all" PrivateAssets="all" />
+    <Reference Include="$(NuGetPackageRoot)SonarAnalyzer.CSharp\$(EmbeddedSonarAnalyzerVersion)\analyzers\SonarAnalyzer.dll" />
+  </ItemGroup>
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/Integration.Vsix.UnitTests.csproj:78]: The tests don't need all of the extra properties/config that the product code does; it just needs to reference the one assembly it tests.

File: src/Integration.Vsix.UnitTests/packages.lock.json [hunk 4]
```diff
@@ -177,12 +177,6 @@
         "resolved": "8.19.0.28253",
         "contentHash": "u7CejPVtZnuZp2nQClK+NCCQCoXNcrCzlAojlNjVUJQVxKwEy7saYygfJWHzG0ihD9oq0tqUeW8K5cYo/N5/ug=="
       },
-      "SonarAnalyzer.VisualBasic": {
-        "type": "Direct",
-        "requested": "[8.19.0.28253, )",
-        "resolved": "8.19.0.28253",
-        "contentHash": "yYnFRXp7N15IVdBX+xWYKwoLtm9pk4WNqj6Aou6TXFXAOgqhbWAX6LSPIWKbwCcy0kV69LcyNoa60i7JeazsmQ=="
-      },
```
Comment (duncanp-sonar) [inline src/Integration.Vsix.UnitTests/packages.lock.json:185]: The tests no longer ref the VB analyzer (which they didn't use anyway).

File: src/Integration.Vsix/Integration.Vsix.csproj [hunk 5]
```diff
@@ -54,6 +54,20 @@
   <!-- Need to reference the Sonar Roslyn analyzers: we need to reference one of the assmblies,
        and embed all of them in the VSIX. -->
   <Import Project="..\EmbeddedSonarAnalyzer.props" />
+  <ItemGroup Label="Embedded SonarLint analyzers">
```
Comment (duncanp-sonar) [inline src/Integration.Vsix/Integration.Vsix.csproj:57]: This section was cut-and-paste without changes from the EmbeddedSonarAnalyzer.props file.

**Review by rita-sonarsource — APPROVED** [review]

**Pull request merged**